src/core/shelf.sol

```diff
@@ -92,42 +111,38 @@ contract Shelf is TitleOwned {
         shelf[loan].initial = principal_;
     }
 
-    // Move the NFT out of the shelf
-    function move(uint loan, address registry_, uint nft_, address to) public owner(loan) {
-        NFTLike(registry_).transferFrom(address(this), to, nft_);
+    function deposit(uint loan, address usr) public owner(loan) {
+        NFTLike(shelf[loan].registry).transferFrom(usr, address(this), shelf[loan].tokenId);
+        pile.borrow(loan, shelf[loan].principal);
+        shelf[loan].principal = 0;
     }
 
-    function release (uint loan, address usr) public owner(loan) {
-        uint debt = pile.debtOf(loan);
-        require(debt == 0, "debt");
-        move(loan, shelf[loan].registry, shelf[loan].tokenId, usr);
-        adjust(loan);
+    // --- Currency actions ---
+    function borrow(uint loan, uint wad) public owner(loan) {
+        debt.accrue(loan);
+        // principal.borrow(loan, wad); TODO: reentrancy
+        debt.inc(loan, wad);
```

Comment:
we need to increase balance here as well for the balance() function of Desk/TrancheManager